File: python_sdk/src/reality_capture/specifications/mesh_sampling.py
from pydantic import BaseModel, Field
from typing import Optional
from enum import Enum


class MeshSamplingInputs(BaseModel):
    meshes: list[str] = Field(description="Reality data Ids of meshes to sample")


class MeshSamplingFormat(Enum):
    OPC = "OPC"
    THREE_D_TILES_PNTS = "3DTilesPnts"
    THREE_D_TILES_GLBC = "3DTilesGlbc"
    LAS = "LAS"
    LAZ = "LAZ"
    E57 = "E57"
    POD = "POD"


class MeshSamplingOptions(BaseModel):
    output_format: Optional[MeshSamplingFormat] = Field(None, description="Output format for the conversion.",
                                                     alias="outputFormat")
    input_crs: Optional[str] = Field(None, description="CRS for the input data", alias="inputCrs")
    output_crs: Optional[str] = Field(None, description="CRS for the output data", alias="outputCrs")
    sampling: Optional[float] = Field(None, description="Sampling value in meter")


class MeshSamplingSpecificationsCreate(BaseModel):
    inputs: MeshSamplingInputs = Field(description="Inputs")
    options: Optional[MeshSamplingOptions] = Field(None, description="Options")


class MeshSamplingSpecifications(BaseModel):
    inputs: MeshSamplingInputs = Field(description="Inputs")
    output: str = Field(description="Reality Data id of the sampled point cloud")
    options: Optional[MeshSamplingOptions] = Field(None, description="Options")

File: typescript/packages/reality-capture/src/specifications/mesh_sampling.ts
import { z } from "zod";

export const MeshSamplingInputsSchema = z.object({
  meshes: z.array(z.string()).describe("Reality data Ids of meshes to sample"),
});
export type MeshSamplingInputs = z.infer<typeof MeshSamplingInputsSchema>;

export enum MeshSamplingFormat {
  OPC = "OPC",
  THREE_D_TILES_PNTS = "3DTilesPnts",
  THREE_D_TILES_GLBC = "3DTilesGlbc",
  LAS = "LAS",
  LAZ = "LAZ",
  E57 = "E57",
  POD = "POD",
}

export const MeshSamplingOptionsSchema = z.object({
  outputFormat: z
    .nativeEnum(MeshSamplingFormat)
    .describe("Output format for the conversion.")
    .optional(),
  inputCrs: z.string().describe("CRS for the input data").optional(),
  outputCrs: z.string().describe("CRS for the output data").optional(),
  sampling: z.number().describe("Sampling value in meter").optional(),
});
export type MeshSamplingOptions = z.infer<typeof MeshSamplingOptionsSchema>;

export const MeshSamplingSpecificationsCreateSchema = z.object({
  inputs: MeshSamplingInputsSchema.describe("Inputs"),
  options: MeshSamplingOptionsSchema.describe("Options").optional(),
});
export type MeshSamplingSpecificationsCreate = z.infer<
  typeof MeshSamplingSpecificationsCreateSchema
>;

export const MeshSamplingSpecificationsSchema = z.object({
  inputs: MeshSamplingInputsSchema.describe("Inputs"),
  output: z.string().describe("Reality Data id of the sampled point cloud"),
  options: MeshSamplingOptionsSchema.describe("Options").optional(),
});
export type MeshSamplingSpecifications = z.infer<
  typeof MeshSamplingSpecificationsSchema
>;
